rust_src/src/strings.rs

```diff
@@ -169,3 +170,35 @@ fn string_to_multibyte(string: LispObject) -> LispObject {
     let _ = string.as_string_or_error();
     unsafe { LispObject::from_raw(c_string_to_multibyte(string.to_raw())) }
 }
+
+/// Return a unibyte string with the same individual chars as STRING.
+/// -If STRING is unibyte, the result is STRING itself.
+/// -Otherwise it is a newly created string, with no text properties,
+/// -where each `eight-bit' character is converted to the corresponding byte.
+/// -If STRING contains a non-ASCII, non-`eight-bit' character,
+/// -an error is signaled.
+#[lisp_fn]
+fn string_to_unibyte(string: LispObject) -> LispObject {
+    let mut string_mut = string;
+    let lispstr = string.as_string_or_error();
+    if lispstr.is_multibyte() {
+        let size = lispstr.len_bytes();
+        let mut buffer: Vec<libc::c_uchar> = Vec::with_capacity(size as usize);
+        let converted_size =
+            multibyte::str_to_unibyte(lispstr.const_data_ptr(), buffer.as_mut_ptr(), size);
+
+        unsafe {
+            if converted_size < size {
+                error(
+                    "Can't convert %dth character to unibyte\0".as_ptr(),
```

Comment:
So after further investigation, it looks like I may not fully understand what that comment is saying. On my system, that macro expands pD to "l", not "t". And in fact hard coding this template to %td instead of %d or %ld leads to the runtime error: 

`(error "Invalid format operation %t")`

I suppose this is something that we may want to define as a rust accessible variable?  I'm not sure the best way to handle this concept.